script/deployPoolWithRenzoStabilityAndMintLiquidity.s.sol

```diff
@@ -0,0 +1,230 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "forge-std/Script.sol";
+import {PositionManager} from "v4-periphery/src/PositionManager.sol";
+import {Hooks} from "v4-core/src/libraries/Hooks.sol";
+import {IHooks} from "v4-core/src/interfaces/IHooks.sol";
+import {HookMiner} from "v4-periphery/src/utils/HookMiner.sol";
+import {PoolKey} from "v4-core/src/types/PoolKey.sol";
+import {CurrencyLibrary, Currency} from "v4-core/src/types/Currency.sol";
+import {Actions} from "v4-periphery/src/libraries/Actions.sol";
+import {LiquidityAmounts} from "v4-core/test/utils/LiquidityAmounts.sol";
+import {TickMath} from "v4-core/src/libraries/TickMath.sol";
+import {IERC20} from "forge-std/interfaces/IERC20.sol";
+import {RenzoStability} from "../src/RenzoStability.sol";
+import {SqrtPriceLibrary} from "../src/libraries/SqrtPriceLibrary.sol";
+import {IRateProvider} from "../src/interfaces/IRateProvider.sol";
+import {LPFeeLibrary} from "v4-core/src/libraries/LPFeeLibrary.sol";
+import {Constants} from "./unichain/Constants.sol";
+import {Config} from "./unichain/Config.sol";
+
+contract CreatePoolAndAddLiquidityScript is Script, Constants, Config {
+    using CurrencyLibrary for Currency;
+
+    /////////////////////////////////////
+    // --- Parameters to Configure --- //
+    /////////////////////////////////////
+
+    // TODO: configure 0 values
+
+    // Hook configuration
+    IRateProvider rateProvider =
+        IRateProvider(0xDb6df3559D2d96985062F0824442550CA7715960);
+    uint24 minFee = 100;
+    uint24 maxFee = 10_000;
+    address ezETH = 0x2416092f143378750bb29b79eD961ab195CcEea5;
+
+    // --- pool configuration --- //
+    // fees paid by swappers that accrue to liquidity providers
+    uint24 lpFee = LPFeeLibrary.DYNAMIC_FEE_FLAG; // Dynamic fee through hook
+    int24 tickSpacing = 1;
+
+    // starting price of the pool, in sqrtPriceX96
+    uint160 startingPrice;
+    IHooks hookContract;
+
+    // --- liquidity position configuration --- //
+    uint256 public token0Amount = 1e18;
+    uint256 public token1Amount = 1e18;
+
+    // range of the position
+    int24 tickLower = -10; // must be a multiple of tickSpacing
+    int24 tickUpper = 10;
```

Comment:
we can do tick derivation as a +/-% of the reference price

https://github.com/saucepoint/jitterbug/blob/94e176790f0be1eecf88bc7b3ae468ad40bce0fa/src/examples/Simple.sol#L60-L67

Reply:
added the utility functions to calculate tick range on the runtime.